### PR TITLE
ERC20Permit Doc

### PR DESCRIPTION
**Pull request by immrsd**
Adds documentation for ERC20Permit [#1103](https://github.com/OpenZeppelin/cairo-contracts/issues/1103)

### CODE REVIEW

File: docs/modules/ROOT/pages/api/erc20.adoc [hunk 1]
```diff
@@ -160,6 +163,48 @@ To create a custom decimals implementation, see {custom-decimals}.
 
 NOTE: This information is only used for _display_ purposes: it in no way affects any of the arithmetic of the contract.
 
+[.contract]
+[[IERC20Permit]]
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/erc20.adoc:167]: This shouldn't be part of the `Core` section, but of an `Extensions` section before `Presets`

File: docs/modules/ROOT/pages/api/erc20.adoc [hunk 2]
```diff
@@ -160,6 +163,48 @@ To create a custom decimals implementation, see {custom-decimals}.
...
+--
+
+[#IERC20Permit-Functions]
+==== Functions
+
+[.contract-item]
+[[IERC20Permit-permit]]
+==== `[.contract-item-name]#++permit++#++(owner: ContractAddress, spender: ContractAddress, amount: u256, deadline: u64, signature: Span<felt252>)++` [.item-kind]#external#
+
+Sets `amount` as the allowance of `spender` over `owner`'s tokens after validating the signature.
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/api/erc20.adoc:192]:
```suggestion
Sets `amount` as the allowance of `spender` over ``owner``'s tokens after validating the signature.
```

In the docsite the current format doesn't show the right way.

File: docs/modules/ROOT/pages/api/erc20.adoc (hunk 1 above)
Comment (immrsd) [reply docs/modules/ROOT/pages/api/erc20.adoc]: Moved to the `Extensions` section

**Review by ericnordelo — APPROVED** [review]
LGTM!

**Pull request merged**